R/comp2long.R

```diff
@@ -1,31 +1,37 @@
 #' Convert composition data to long format
 #'
-#' Convert data frame containing length or generalized size composition data
-#' from wide to long format.
+#' Convert age, length, or generalized size composition data from wide to long
+#' format.
 #'
-#' @param x A data frame containing length or generalized size composition data,
-#'        or a list containing such a data frame.
+#' @param x A data frame containing age, length, generalized size composition
+#'        data, or a list containing such a data frame.
+#' @param ... Passed to \code{age2long} or \code{size2long}.
+#' @param expand Whether to repeat recurring entries in the resulting data
+#'        frame, so that the \code{freq} column is \code{1} for every entry.
+#' @param zero Whether zero frequencies should be included in the output.
 #' @param measure An optional string indicating the type of measurement, e.g.,
 #'        \code{"weight"}.
-#' @param zero Whether zero frequencies should be included in the output.
 #'
 #' @details
-#' If \code{x} is not a data frame, the function will look for data frames
-#' called \code{x[["lencomp"]]} or \code{x[["dat"]][["lencomp"]]}.
+#' The \code{age2long} function converts age compositions and \code{size2long}
+#' converts length or generalized size compositions. Alternatively, the wrapper
+#' function \code{comp2long} converts any composition data and will call either
+#' \code{age2long} or \code{size2long}, depending on the data type of \code{x}.
 #'
-#' @return
-#' Data frame containing the columns
-#' \preformatted{year  month  fleet  sex  part  Nsamp  length  freq}
-#' for length composition data, or
-#' \preformatted{method  year  month  fleet  sex  part  Nsamp  size  freq}
-#' for generalized size composition data.
+#' If \code{x} is not a data frame, the function will look for \code{lencomp}
+#' inside the list.
 #'
-#' The second to last column can be named explicitly via the measure argument.
+#' The \code{expand = TRUE} option is mainly useful for accessing conditional
+#' age-at-length, e.g., to produce a data frame that has the same number of rows
+#' as the number of otoliths.
```

Comment:
`otoliths` isn't in WORDLIST. You can update it via `spelling::update_wordlist()`